### PR TITLE
Isolate fetching cards to one place on the client

### PR DESCRIPTION
**Pull request by kberg**
This will make it far easier to find a replacement that doesn't import actual card instances. Plus by doing this all the code uses the same API, making things so much simpler! Look at how small `AllCards.ts` is. I already see new minor things that can change and improve.

### CODE REVIEW

File: src/client/components/SelectInitialCards.vue [hunk 1]
```diff
@@ -67,11 +68,12 @@ export default Vue.extend({
...
       let result = 0;
       for (const prelude of this.selectedPreludes) {
-        const card = new CardFinder().getPreludeByName(prelude);
-        if (card === undefined) {
+        const cam = getCard(prelude);
+        if (cam === undefined) {
           throw new Error(`Prelude ${prelude} not found`);
         }
-        result += card?.startingMegaCredits ?? 0;
+        const card = cam.card as PreludeCard;
```
Comment (bafolts) [inline src/client/components/SelectInitialCards.vue:75]: Should we check if this is a prelude card? If a project card ended up in `selectedPreludes` this may try to load it. Prior to this we only looked through prelude cards. Now that we look through project cards we could get a project card. Using `instanceof` should check that it is a `PreludeCard`. I think if you check `instanceof` typescript may also then let you use it as `PreludeCard` without casting.

File: src/client/components/create/CorporationsFilter.vue [hunk 2]
```diff
@@ -86,25 +76,35 @@ export default Vue.extend({
...
   data() {
-    const cardsByModuleMap: Map<GameModule, Array<CardName>> =
-            new Map(ALL_CARD_MANIFESTS.map((m) => [m.module, corpCardNames(m.module)]));
+    const cardsByModule: MultiMap<GameModule, CardName> = new MultiMap();
+    getCards(byType(CardType.CORPORATION)).forEach((cam) => {
+      if (cam.card.name !== CardName.BEGINNER_CORPORATION) {
+        cardsByModule.set(cam.module, cam.card.name);
+      }
+    });
+    console.log(iterator(cardsByModule));
```
Comment (bafolts) [inline src/client/components/create/CorporationsFilter.vue:90]: Assuming this was for debugging. Is if iterator isn't used anywhere else I would remove it. If keeping iterator I would consider a different name. Javascript has iterators but the function named iterator returns an Array which could be confusing.

File: src/client/components/create/CorporationsFilter.vue [hunk 3]
```diff
@@ -116,9 +116,9 @@ export default Vue.extend({
       return [];
     },
     getItemsByGroup(group: string): Array<CardName> {
-      if (group === 'All') return allItems.slice();
+      if (group === 'All') return Array.from(this.cardsByModule.values()).slice();
```
Comment (bafolts) [inline src/client/components/create/CorporationsFilter.vue:119]: The `slice` here was being used to create a copy of `allItems` so that array wouldn't be modified. Using `Array.from` should create a new `Array` so the usage of `slice` shouldn't be needed. Could also see if this array were marked `ReadonlyArray<CardName>` if anything is trying to modify this. If nothing is trying to modify then could keep it as `ReadonlyArray`.

File: src/client/components/create/CreateGameForm.vue [hunk 4]
```diff
@@ -415,6 +415,7 @@
                   v-bind:promoCardsOption="promoCardsOption"
                   v-bind:communityCardsOption="communityCardsOption"
                   v-bind:moonExpansion="moonExpansion"
+                  v-bind:pathfindersExpansion="pathfindersExpansion"
```
Comment (bafolts) [inline src/client/components/create/CreateGameForm.vue:418]: Was this change intentional? Seems unrelated to the other changes.

**Review by bafolts — APPROVED** [review]

File: src/client/components/create/CorporationsFilter.vue (hunk 2 above)
Comment (kberg) [reply src/client/components/create/CorporationsFilter.vue:90]: Oh yeah it's gotta go. Good catch.

File: src/client/components/create/CreateGameForm.vue (hunk 4 above)
Comment (kberg) [reply src/client/components/create/CreateGameForm.vue:418]: You know, you're right? It is unrelated. But somehow it was absent. So I tossed it in.

File: src/client/components/create/CorporationsFilter.vue (hunk 3 above)
Comment (kberg) [reply src/client/components/create/CorporationsFilter.vue:119]: Good call removing `slice.`

Let's hold off on the `ReadonlyArray` stuff which is also a little more powerful thanks to recent TS versions: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html I think that kind of stuff can have a surprise effect so I'd rather be careful.

File: src/client/components/SelectInitialCards.vue (hunk 1 above)
Comment (kberg) [reply src/client/components/SelectInitialCards.vue:75]: Good question. Actually I'm going to advise no because `PreludeCard` will eventually go away.  The client code will no longer rely on `ICard` or any `ICard`-related stuff. That's how we're gonna get rid of all the server code on the client.
